mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -52,16 +52,18 @@ public AccountManagerUtil(Context context, String accountName, boolean sharedSto
             //Attempt to retrieve the account
             Account[] accounts = mAccountManager.getAccountsByType(accountType);
             for (Account account : accounts) {
-                if (accountName.equals(account.name)) {
+                if (accountName != null &&accountName.equals(account.name)) {
                     String password = mAccountManager.getPassword(account);
                     String savedPassword = identifier.toString();
-                    if (password.equals(savedPassword)) {
+                    if (password != null && password.equals(savedPassword)) {
                         mAccount = account;
                     }else {
                         // - case migration from old AccountManagerStoreDataSource
                         mAccount = null;
                         identifier = new SharedStorageIdentifier();
                     }
+                } else {
+                    throw new IllegalArgumentException("Invalid parameters, Account name cannot be null");
```

Comment:
[optional], "Account name cannot be null" is sufficient here.

Reply:
@graju256 @zakirjt  Added some logs to capture some details inside the catch block. Can you please review that.